### PR TITLE
#21209: Skeleton for Artery docs

### PR DESCRIPTION
**Pull request by drewhk**
Refs #21209

### CODE REVIEW

File: akka-docs/rst/scala/remoting-artery.rst [hunk 1]
```diff
@@ -0,0 +1,469 @@
...
+
+In order to remoting to work properly, where each system can send messages to any other system on the same network
+(for example a system forwards a message to a third system, and the third replies directly to the sender system)
+it is essential for every system to have a *unique, globally reachable* address and port. This address is part of the
+unique name of the system and will be used by other systems to open a connection to it and send messages. This means
+that if a host has multiple names (different DNS records pointing to the same IP address) then only one of these
+can be *canonical*. If a message arrives to a system but it contains a different hostname than the expected canonical
+name then the message will be dropped. If multiple names for a system would be allowed, then equality checks among
+:class:`ActorRef` instances would no longer to be trusted and this would violate the fundamental assumption that
+an actor has a globally unique reference on a given network.
```
Comment (johanandren) [inline akka-docs/rst/scala/remoting-artery.rst]: 👍

File: akka-docs/rst/scala/remoting-artery.rst [hunk 2]
```diff
@@ -0,0 +1,469 @@
...
+the system will start listening on a provided network address and also gains the ability to connect to other
+systems through the network. From the application's perspective there is no API difference between local or remote
+systems, :class:`ActorRef` instances that point to remote systems look exactly the same as local ones: they can be
+sent messages to, watched, etc.
+Every :class:`ActorRef` contains hostname and port information and can be passed around even on the network. This means
+that on a network every :class:`ActorRef` is a unique identifier of an actor on that network.
+
+Remoting is not a server-client technology. All systems using remoting can contact any other system on the network
+if they possess an :class:`ActorRef` pointing to those system. This means that every system that is remoting enabled
+acts as a "server" to which arbitrary systems on the same network can connect to.
```
Comment (johanandren) [inline akka-docs/rst/scala/remoting-artery.rst]: Great spelling this out explicitly.

File: akka-docs/rst/scala/remoting-artery.rst [hunk 3]
```diff
@@ -0,0 +1,469 @@
...
+.. note::
+
+  Message sends to actors that are actually in the sending actor system do not
+  get delivered via the remote actor ref provider. They're delivered directly,
+  by the local actor ref provider.
+
+  Aside from providing better performance, this also means that if the hostname
+  you configure remoting to listen as cannot actually be resolved from within
+  the very same actor system, such messages will (perhaps counterintuitively)
+  be delivered just fine.
```
Comment (johanandren) [inline akka-docs/rst/scala/remoting-artery.rst]: A bit weird with two notes after each other, maybe the first one should just be a regular paragraph?

File: akka-docs/rst/scala/remoting-artery.rst [hunk 4]
```diff
@@ -0,0 +1,469 @@
...
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+If you want to use the creation functionality in Akka remoting you have to further amend the
+``application.conf`` file in the following way (only showing deployment section)::
+
+  akka {
+    actor {
+      deployment {
+        /sampleActor {
+          remote = "akka.tcp://sampleActorSystem@127.0.0.1:2553"
```
Comment (johanandren) [inline akka-docs/rst/scala/remoting-artery.rst]: `akka://`

File: akka-docs/rst/scala/remoting-artery.rst [hunk 5]
```diff
@@ -0,0 +1,469 @@
...
+  ``LocalScope`` are exempt from this check.
+
+.. note::
+
+  You can use asterisks as wildcard matches for the actor paths, so you could specify:
+  ``/*/sampleActor`` and that would match all ``sampleActor`` on that level in the hierarchy.
+  You can also use wildcard in the last position to match all actors at a certain level:
+  ``/someParent/*``. Non-wildcard matches always have higher priority to match than wildcards, so:
+  ``/foo/bar`` is considered **more specific** than ``/foo/*`` and only the highest priority match is used.
+  Please note that it **cannot** be used to partially match section, like this: ``/foo*/bar``, ``/f*o/bar`` etc.
```
Comment (johanandren) [inline akka-docs/rst/scala/remoting-artery.rst]: Another double-note

File: akka-docs/rst/scala/remoting-artery.rst [hunk 6]
```diff
@@ -0,0 +1,469 @@
...
+
+.. image:: ../images/association_lifecycle.png
+   :align: center
+   :width: 620
+
+Each link with a remote system can be in one of the four states as illustrated above. Before any communication
+happens with a remote system at a given ``Address`` the state of the association is ``Idle``. The first time a message
+is attempted to be sent to the remote system or an inbound connection is accepted the state of the link transitions to
+``Active`` denoting that the two systems has messages to send or receive and no failures were encountered so far.
+When a communication failure happens and the connection is lost between the two systems the link becomes ``Gated``.
```
Comment (johanandren) [inline akka-docs/rst/scala/remoting-artery.rst]: Do we still have something like the `Idle` state of a connection with artery?

Comment (johanandren) [inline akka-docs/rst/scala/remoting-artery.rst]: `Gating` is also not here anymore?

File: akka-docs/rst/scala/remoting-artery.rst (hunk 4 above)
Comment (drewhk) [inline akka-docs/rst/scala/remoting-artery.rst]: ah, thanks

File: akka-docs/rst/scala/remoting-artery.rst (hunk 6 above)
Comment (drewhk) [inline akka-docs/rst/scala/remoting-artery.rst]: We likely need a new lifecycle section.

**Comment by johanandren** [comment]
LGTM (comments are nicely fixable after merging I think)

**Comment by drewhk** [comment]
Ok, so should I merge? Then we can coordinate on the ticket.

**Pull request closed without merging**